nsls2forge_utils/auto_tick.py

```diff
@@ -0,0 +1,627 @@
+                    }
+                else:
+                    if migrator_uid:
+                        # On successful PR add to our counter
+                        good_prs += 1
+                finally:
+                    # Write graph partially through
+                    if not dry_run:
+                        dump_graph(mctx.graph)
+
+                    eval_cmd(f"rm -rf {mctx.rever_dir}/*")
+                    logger.info(os.getcwd())
+                    for f in glob.glob("/tmp/*"):
+                        if f not in temp:
+                            eval_cmd(f"rm -rf {f}")
+
+    if not dry_run:
+        logger.info(
+            "API Calls Remaining: %d",
+            mctx.gh.rate_limit()["resources"]["core"]["remaining"],
+        )
+    logger.info("Done")
+
+
+def status_report():
+    '''
+    Write out the status of current/recent migrations and their
+    pull requests on GitHub.
+
+    Only works for Version migrations at the moment.
+    '''
+    print('Determining current status of migrations...')
+    mctx, *_, migrators = initialize_migrators()
+    if not os.path.exists("./status"):
+        os.mkdir("./status")
+
+    for migrator in migrators:
+        if isinstance(migrator, Version):
+            write_version_migrator_status(migrator, mctx)
+
+    lst = [
+        k
+        for k, v in mctx.graph.nodes.items()
+        if len(
+            [
+                z
+                for z in v.get("payload", {}).get("PRed", [])
+                if z.get("PR", {}).get("state", "closed") == "open"
+                and z.get("data", {}).get("migrator_name", "") == "Version"
+            ],
+        )
+        >= Version.max_num_prs
+    ]
+    with open("./status/could_use_help.json", "w") as f:
+        json.dump(
+            sorted(
+                lst,
+                key=lambda z: (len(nx.descendants(mctx.graph, z)), lst),
+                reverse=True,
+            ),
+            f,
+            indent=2,
+        )
+    print('Statuses have been placed in ./status')
+
+
+def clean(include=None, exclude=None, yes=False):
+    '''
+    Cleans the current directory for a fresh run of the
+    graph building and auto-tick process.
+
+    Parameters
+    ----------
+    include: list, optional
+        Files to be removed (default is all files required
+        by the bot except names.txt)
+    exclude: list, optional
+        Files to keep if already included either by default
+        or by user
+    yes: bool, optional
+        Automatically proceed with deletion if True
+    '''
+    to_be_removed = [
+        './dask-worker-space/*',
+        './node_attrs/*',
+        './feedstocks/*',
+        './pr_json/*',
+        './versions/*',
+        './status/*',
+        'graph.json'
+    ]
+    if include is not None:
+        to_be_removed += include
+    if exclude is not None:
+        to_be_removed = [p for p in to_be_removed if p not in exclude]
```

Comment:
Probably include/exclude logic can be done with Python's [`set`](https://docs.python.org/2/library/sets.html)s, such as:
```py
In [31]: a = set([1, 2, 3])

In [32]: include = set([3, 4, 8])

In [33]: exclude = set([ 3, 5])

In [34]: a.union(include)
Out[34]: {1, 2, 3, 4, 8}

In [35]: a.difference(exclude)
Out[35]: {1, 2}
```

Reply:
So something like this would work better:

```
to_be_removed = set(to_be_removed)
if include is not None:
    include = set(include)
    to_be_removed = to_be_removed.union(include)
if exclude is not None:
    exclude = set(exclude)
    to_be_removed = to_be_removed.difference(exclude)
to_be_removed = list(to_be_removed)
```